src/System.DirectoryServices.Protocols/tests/BerConversionExceptionTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace System.DirectoryServices.Protocols.Tests
+{
+    public class BerConversionExceptionTests
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            var exception = new BerConversionException();
+            Assert.NotEmpty(exception.Message);
+            Assert.Null(exception.InnerException);
+        }
+
+        [Fact]
+        public void Ctor_Message()
+        {
+            var exception = new BerConversionException("message");
+            Assert.Equal("message", exception.Message);
+            Assert.Null(exception.InnerException);
+        }
+
+        [Fact]
+        public void Ctor_Message_InnerException()
+        {
+            var innerException = new Exception();
+            var exception = new BerConversionException("message", innerException);
+            Assert.Equal("message", exception.Message);
+            Assert.Same(innerException, exception.InnerException);
+        }
+
+        [Fact]
+        public void SubClass_Deserialize_ThrowsPlatformNotSupportedException()
+        {
+            using (var stream = new MemoryStream())
+            {
+                var formatter = new BinaryFormatter();
+                formatter.Serialize(stream, new SubException());
```

Comment:
`BerConversionException` et al. don't have a `[Serializable]` attribute so we get an error trying to serialize them - we want to deserialize it in these tests